runtime/cudaq/algorithms/state.cpp

```diff
@@ -74,4 +75,30 @@ double state::overlap(state &other) {
   // return the overlap
   return sum;
 }
+
+bool state::is_density_matrix() {
+  auto &[shape, stateData] = data;
+  if (shape.size() == 1) {
+    return false;
+  } else {
+    assert(shape.size() > 1);
+    return true;
+  }
+}
+
+template <>
```

Comment:
With the update to `get_data() : vector<complex>*` you'll be able to remove these.